primitives/tf_encrypted/primitives/paillier/primitives.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+import tensorflow as tf
+
+import tf_big
+
+tf_big.set_secure_default(True)
+
+
+class EncryptionKey:
+    def __init__(self, n):
+        n = tf_big.convert_to_tensor(n)
+
+        self.n = n
+        self.nn = n * n
+        self.g = 1 + n
+
+
+class DecryptionKey:
+    def __init__(self, p, q):
+        p = tf_big.convert_to_tensor(p)
+        q = tf_big.convert_to_tensor(q)
+
+        n = p * q
+        self.n = n
+        self.nn = n * n
+        self.g = 1 + n
+
+        order_of_n = (p - 1) * (q - 1)
+        self.d1 = order_of_n
+        self.d2 = tf_big.inv(order_of_n, n)
+        self.e = tf_big.inv(n, order_of_n)
+
+
+def gen_keypair():
+    # TODO
```

Comment:
should sample two primes here; see [libpaillier](https://github.com/snipsco/paillier-libraries-benchmarks/blob/master/c-libpaillier/libpaillier-0.8/paillier.c#L57-L114)